src/Features/Core/Portable/ConvertAutoPropertyToFullProperty/AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,220 @@
+        internal abstract string GetUniqueName(string fieldName, IPropertySymbol property);
+        internal abstract SyntaxNode GetInitializerValue(SyntaxNode property);
+        internal abstract SyntaxNode GetPropertyWithoutInitializer(SyntaxNode property);
+        internal abstract Task<(SyntaxNode newGetAccessor, SyntaxNode newSetAccessor)> GetNewAccessorsAsync(
+            Document document, SyntaxNode property, string fieldName, SyntaxGenerator generator, 
+            CancellationToken cancellationToken);
+        internal abstract SyntaxNode GetTypeBlock(SyntaxNode syntaxNode);
+
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(context.Span.Start);
+
+            var property = GetProperty(token);
+            if (property == null)
+            {
+                return;
+            }
+
+            if (!(await IsValidAutoProperty(property, document, cancellationToken).ConfigureAwait(false)))
+            {
+                return;
+            }
+
+            context.RegisterRefactoring(
+                new ConvertAutoPropertyToFullPropertyCodeAction(
+                    FeaturesResources.Convert_to_full_property,
+                    c => ExpandToFullPropertyAsync(
+                        document,
+                        property,
+                        root,
+                        cancellationToken)));
+        }
+
+        internal async Task<bool> IsValidAutoProperty(
+            SyntaxNode property, Document document, CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            var propertySymbol = semanticModel.GetDeclaredSymbol(property) as IPropertySymbol;
+
+            var members = propertySymbol.ContainingType.GetMembers()
+                .Where(n => n.Kind == SymbolKind.Field);
+
+            IFieldSymbol fieldSymbol;
+            foreach (var member in members)
+            {
+                fieldSymbol = (IFieldSymbol)member;
+                if (fieldSymbol.AssociatedSymbol?.Name == propertySymbol.Name)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private async Task<Document> ExpandToFullPropertyAsync(
+            Document document,
+            SyntaxNode property,
+            SyntaxNode root,
+            CancellationToken cancellationToken)
+        {
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var propertySymbol = semanticModel.GetDeclaredSymbol(property) as IPropertySymbol;
```

Comment:
Get the property symbol once at the top level and pass to all your methods that use it.